DEVELOPING.md

```diff
@@ -0,0 +1,60 @@
+# Getting started
+
+Clone the project and `cd` into it:
+```
+git clone git@github.com:jsforce/jsforce.git
+cd jsforce
```

Comment:
Is the plan to have the v3 branch to eventually become `main`? If not, you could add switching branches here

Reply:
> Is the plan to have the v3 branch to eventually become main?

yes.